### ISSUE TITLE
histogram2d bug (or wrong documentation)

### ISSUE THREAD

**Issue by afkowalski** [numpy/numpy#18929]
I wasn't sure if this is a bug or a documentation issue.  It is not stated that to use pcolormesh after invoking histogram2d, you have to transpose the returned histogram!  I see that H.T is in one of the examples, but it's certainly not stated that you MUST do this.  The x and y mapping to row and column in python is a headache, and it would be helpful if this issue was explicitly stated in the documentation for histogram2d.  Thank you for your consideration.

### Reproducing code example:

```python
import numpy as np
<< your code here >>
```

**Comment by eric-wieser** [comment]
histogramdd does not do this transposition, for what it's worth.

**Comment by marysia** [comment]
I'm working on this.

**Comment by marysia** [comment]
The example code for `NonUniformImage` in the documentation throws an error at the `ax.images.append(im)` line: 

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/IPython/core/formatters.py in __call__(self, obj)
    339                 pass
    340             else:
--> 341                 return printer(obj)
    342             # Finally look for special method names
    343             method = get_real_method(obj, self.print_method)

~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/IPython/core/pylabtools.py in <lambda>(fig)
    246 
    247     if 'png' in formats:
--> 248         png_formatter.for_type(Figure, lambda fig: print_figure(fig, 'png', **kwargs))
    249     if 'retina' in formats or 'png2x' in formats:
    250         png_formatter.for_type(Figure, lambda fig: retina_figure(fig, **kwargs))

~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/IPython/core/pylabtools.py in print_figure(fig, fmt, bbox_inches, **kwargs)
    130         FigureCanvasBase(fig)
    131 
--> 132     fig.canvas.print_figure(bytes_io, **kw)
    133     data = bytes_io.getvalue()
    134     if fmt == 'svg':

~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/matplotlib/backend_bases.py in print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, bbox_inches, pad_inches, bbox_extra_artists, backend, **kwargs)
   2232             if bbox_inches:
   2233                 if bbox_inches == "tight":
-> 2234                     bbox_inches = self.figure.get_tightbbox(
   2235                         renderer, bbox_extra_artists=bbox_extra_artists)
   2236                     if pad_inches is None:

~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/matplotlib/figure.py in get_tightbbox(self, renderer, bbox_extra_artists)
   1634 
   1635         for a in artists:
-> 1636             bbox = a.get_tightbbox(renderer)
   1637             if bbox is not None and (bbox.width != 0 or bbox.height != 0):
   1638                 bb.append(bbox)

~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/matplotlib/axes/_base.py in get_tightbbox(self, renderer, call_axes_locator, bbox_extra_artists, for_layout_only)
   4482                     # this artist
   4483                     continue
-> 4484             bbox = a.get_tightbbox(renderer)
   4485             if (bbox is not None
   4486                     and 0 < bbox.width < np.inf

~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/matplotlib/artist.py in get_tightbbox(self, renderer)
    304             The enclosing bounding box (in figure pixel coordinates).
    305         """
--> 306         bbox = self.get_window_extent(renderer)
    307         if self.get_clip_on():
    308             clip_box = self.get_clip_box()

~/.pyenv/versions/3.8.3/lib/python3.8/site-packages/matplotlib/image.py in get_window_extent(self, renderer)
    911 
    912     def get_window_extent(self, renderer=None):
--> 913         x0, x1, y0, y1 = self._extent
    914         bbox = Bbox.from_extents([x0, y0, x1, y1])
    915         return bbox.transformed(self.axes.transData)

TypeError: cannot unpack non-iterable NoneType object
```

**Comment by rossbar** [comment]
> The example code for NonUniformImage in the documentation throws an error at the ax.images.append(im) line:

Interesting... I'm having trouble reproducing this - how were you invoking the doctests?

**Comment by marysia** [comment]
We didn't invoke the doctests - we wanted to run the code example from the documentation, so copy-pasted the code (with the latest version of numpy and matplotlib).

**Comment by mattip** [comment]
Closed by #18959
